File: DAL/Entities/LinkedDatabase.cs
using DAL.Entities.Enums;
using System.ComponentModel.DataAnnotations.Schema;

namespace DAL.Entities;

public class LinkedDatabase: BaseEntity
{
    public Guid ClientId { get; set; }

    [ForeignKey(nameof(ClientId))]
    public Client? Client { get; set; }

    public DatabaseType DatabaseType { get; set; }

    // Configuration of additional tables

    public bool HasViews { get; set; }
    public bool HasRatings { get; set; }

    public string? DatabaseConfigurationId { get; set; }
    public string? DatabaseLink { get; set; }

    public string? Structure { get; set; }
}

File: RecifyWeb/src/app/models/linkedDatabase.ts
export class LinkedDatabase {
    id: string;
    clientId: string;
    linkedDatabaseId: string;
    databaseType: number;
    hasViews: boolean;
    hasRatings: boolean;
    databaseConfigurationId: string = '';
    databaseLink: string = '';
    structure: string = '';

    constructor(
        id: string,
        clientId: string,
        linkedDatabaseId: string,
        databaseType: number,
        hasViews: boolean,
        hasRatings: boolean,
        databaseConfigurationId: string = '',
        databaseLink: string = '',
        structure: string = ''
    ) {
        this.id = id;
        this.clientId = clientId;
        this.linkedDatabaseId = linkedDatabaseId;
        this.databaseType = databaseType;
        this.hasViews = hasViews;
        this.hasRatings = hasRatings;
        this.databaseConfigurationId = databaseConfigurationId;
        this.databaseLink = databaseLink;
        this.structure = structure;
    }
}